src/reconcile/rec_write.c

```diff
@@ -2317,12 +2936,19 @@ __rec_split_discard(WT_SESSION_IMPL *session, WT_PAGE *page)
          */
         if (multi->addr.block_cookie != NULL) {
             WT_RET(__wt_btree_block_free(
-              session, multi->addr.block_cookie, multi->addr.block_cookie_size));
+              session, multi->addr.block_cookie, multi->addr.block_cookie_size, false));
             __wt_free(session, multi->addr.block_cookie);
         }
+
+        multi->block_meta.page_id = WT_BLOCK_INVALID_PAGE_ID;
     }
     __wt_free(session, mod->mod_multi);
     mod->mod_multi_entries = 0;
+    mod->rec_result = 0;
+
+    /* Also reset the page's latest LSN, so that it can be safely discarded. */
+    /* TODO: Is this necessary? */
+    page->rec_lsn_max = WT_DISAGG_LSN_NONE;
```

Comment:
I think this is unnecessary. It will be updated at the end of the reconciliation anyway.